### PR TITLE
HWKALERTS-181 Add clustering information on status endpoint

### PR DESCRIPTION
**Pull request by lucasponce**
Refactor Status logic to have its own service

Basically two things on this PR:
- To have a proper StatusService on the impl, before that the status endpoint was calculated by a simple DefinitionsService call, StatusService might hide and encapsulate other future complex checks.
- Add clustering info on status endpoint.

### CODE REVIEW

File: hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/services/StatusService.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+     * @return true if system is running on a distributed scenario.
+     *         false if system is running on a standalone scenario.
+     */
+    boolean isDistributed();
+
+    /**
+     * Show additional information about distributed status.
+     * In distributed scenarios
+     *  - getDistributedStatus().get("currentNode") will store a string with the identifier of the current node
+     *  - getDistributedStatus().get("members") will store a string with a list comma identifiers of the nodes of the topology
```
Comment (jshaughn) [inline hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/services/StatusService.java]: In the above two comment lines did you mean "will store"  or "will return"?

File: hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/impl/PartitionManagerImpl.java [hunk 2]
```diff
@@ -183,6 +186,19 @@ public boolean isDistributed() {
         return distributed;
     }
 
+    @Override
+    public Map<String, String> getStatus() {
+        if (distributed) {
+            /*
+                Members are dynamic, so on each status request this info is refreshed
+             */
+            status.put("members", cacheManager.getMembers().stream()
```
Comment (jshaughn) [inline hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/impl/PartitionManagerImpl.java]: Block comments are so C :)   Aside from a preference for //, this is pretty cool, it looks like you are able to actually get member info, and not just a member count, that will be very useful, I think.

File: hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/service/PartitionManager.java [hunk 3]
```diff
@@ -72,6 +73,18 @@
     boolean isDistributed();
 
     /**
+     * Show additional information about partition status.
+     * In distributed scenarios
+     *  - getStatus().get("currentNode") will store a string with the identifier of the current node
+     *  - getStatus().get("members") will store a string with a list comma identifiers of the nodes of the topology
```
Comment (jshaughn) [inline hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/service/PartitionManager.java]: same comment about "will store".  Actually, you probably only need the comments here in the interface, not in the impl.

File: hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/services/StatusService.java (hunk 1 above)
Comment (lucasponce) [inline hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/services/StatusService.java]: Well, I guess I translated here, will return is better.

File: hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/impl/PartitionManagerImpl.java (hunk 2 above)
Comment (lucasponce) [inline hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/impl/PartitionManagerImpl.java]: I prefer the C syntax for comments, it makes comment more clear for me :-).
Having this extra space makes me to pay better attention to it.

File: hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/service/PartitionManager.java (hunk 3 above)
Comment (jshaughn) [inline hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/service/PartitionManager.java]: certainly it's harmless.  IDE typically just grabs from the super, so it saves you from having to have them in both places.  Not a big deal, just a note.

File: hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/services/StatusService.java (hunk 1 above)
Comment (stefannegrea) [inline hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/services/StatusService.java]: Nitpicking here .... but I would drop `will`.

**Pull request merged**